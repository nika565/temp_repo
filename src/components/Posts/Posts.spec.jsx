const { render, screen } = require("@testing-library/react");
import { Posts } from  './index'

const props = {
    posts: [
        {
            id: 1,
            title: 'title 1',
            body: 'body 1',
            cover: 'img/img1.png'
        },

        {
            id: 2,
            title: 'title 2',
            body: 'body 2',
            cover: 'img/img2.png'
        },

        {
            id: 3,
            title: 'title 3',
            body: 'body 3',
            cover: 'img/img3.png'
        },
    ]
}

describe('<Posts />', () => {

    it('should render posts', () => {

        // renderizando a o componente e passando dados mockados
        render(<Posts {...props}/>);

        // Selecionando os titulos do meu componente
        expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);

        // Selecionando as imagens do meu componente
        expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);
       
        // Selecionando os parágrafos do meu componente
        expect(screen.getAllByText(/body/i)).toHaveLength(3);

        // Pegando uma imagem específica e verificando o seu atributo
        expect(screen.getByRole('img', { name: 'title 3' })).toHaveAttribute('src', 'img/img3.png');

    });

    it('should not render posts', () => {

        render(<Posts/>);

        expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument();

    })

    it('should match snapshot', () => {

        const { container } = render(<Posts {...props}/>);

        expect(container.firstChild).toMatchSnapshot();

    })

});