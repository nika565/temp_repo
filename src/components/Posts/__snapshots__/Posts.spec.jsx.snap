// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
         
        1
        - 
        title 1
      </h2>
      <p>
        body 1
      </p>
      <p>
        Oi
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
         
        2
        - 
        title 2
      </h2>
      <p>
        body 2
      </p>
      <p>
        Oi
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
         
        3
        - 
        title 3
      </h2>
      <p>
        body 3
      </p>
      <p>
        Oi
      </p>
    </div>
  </div>
</div>
`;
